src/main/java/co/cask/hydrator/plugin/batch/sink/FailPipelineSink.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.Output;
+import co.cask.cdap.api.data.batch.OutputFormatProvider;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import org.apache.hadoop.mapreduce.lib.output.NullOutputFormat;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * CDAP Fail Pipeline batch Sink.
+ */
+@Plugin(type = "batchsink")
+@Name("FailPipeline")
+@Description("Fails the pipeline if any record flows to this sink.")
+public class FailPipelineSink extends BatchSink<StructuredRecord, Void, Void> {
```

Comment:
extend the reference batch Sink instead of the batch sink, That way we will have a config property of plugin reference batch name.